pygmt/helpers/utils.py

```diff
@@ -185,11 +200,19 @@ def build_arg_string(kwargs):
         if kwargs[key] is None or kwargs[key] is False:
             pass  # Exclude arguments that are None and False
         elif is_nonstr_iter(kwargs[key]):
-            gmt_args.extend(f"-{key}{value}" for value in kwargs[key])
+            for value in kwargs[key]:
+                _value = str(value).replace(" ", r"\040")
+                gmt_args.append(rf"-{key}{_value}")
         elif kwargs[key] is True:
             gmt_args.append(f"-{key}")
         else:
-            gmt_args.append(f"-{key}{kwargs[key]}")
+            if key != "J":  # non-projection parameters
+                _value = str(kwargs[key]).replace(" ", r"\040")
+            else:
+                # special handling if key == "J" (projection)
+                # remove any spaces in PROJ4 string
+                _value = str(kwargs[key]).replace(" ", "")
+            gmt_args.append(rf"-{key}{_value}")
```

Comment:
I know almost nothing about PROJ4, but does a PROJ4 string always contains `+proj=xxx` or `EPSG`? If yes, then we can simplify the if-test to something like:
```
            if key != "J" or (key == "J" and "+proj" not in value and "EPSG" not in value):
                _value = str(kwargs[key]).replace(" ", r"\040")
            else:
                # special handling if -J + PROJ4 string
                # remove any spaces in PROJ4 string
                _value = str(kwargs[key]).replace(" ", "")
            gmt_args.append(rf"-{key}{_value}")
```
The motivation is, since PROJ4 is not commonly used, it seems a waste of time to do call the `replace` method for GMT-style -J.

Reply:
Not sure if they all have either `+proj=xxx` or `EPSG` (see https://proj.org/operations/projections/index.html), but they probably do? I know that there's `+init=EPSG:xxx` which works when I tried here, but that's actually deprecated syntax (see https://pyproj4.github.io/pyproj/stable/gotchas.html#init-auth-auth-code-should-be-replaced-with-auth-auth-code) and not probably something we want to allow.

Also, not sure if 2 extra if-checks saves time, considering that either way, a `str.replace` still needs to happen for the space character - either to `\040` or to nothing ``.